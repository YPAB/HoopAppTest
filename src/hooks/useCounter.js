import { useState } from "react"

export const useCounter = (initialState = 10) => {
    const [counter,setCounter] = useState(initialState);

    const increment = () => {
        setCounter( counter + 1 );
    }

    const decrement = () => {
        setCounter( counter - 1 );
    }

    const reset = () => {
        setCounter( initialState );
    }

    //retornamos la logica de nuestro hook
    return {
        counter,
        increment,
        decrement,
        reset
    };

}