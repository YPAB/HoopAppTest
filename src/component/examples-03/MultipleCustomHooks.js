import React from "react";
import { useCounter } from "../../hooks/useCounter";
import { useFetch } from "../../hooks/useFetch";
import '../useEffect-02/effects.css';

export const MultipleCustomHooks = () => {

    const {counter,increment} = useCounter(1);

    const url = `https://www.breakingbadapi.com/api/quotes/${ counter }`;

    const { data,loading,error } = useFetch(url);

    const { author,quote } = !!data && data[0]; //si existe la data entonces extrae la posicion cero de la data.


    return (
        <div>
            <h1>
                Breaking Bad !!
            </h1>
            <hr/>

            {
                loading
                ?
                (
                    <div className='alert alert-info text-center'>
                        Cargando ...

                     </div>

                )
                :

                (
                    <blockquote className='blockquote text-end'>
                        <p className='mb-0'>
                            { quote }
                        </p>
                        <br/>
                        <footer className='blockquote-footer'>
                            { author }
                        </footer>
                    </blockquote>


                )
            }

            

        <button className='btn btn-info' onClick={ increment }>

            Siguiente Frase

        </button>
        </div>
    )


}