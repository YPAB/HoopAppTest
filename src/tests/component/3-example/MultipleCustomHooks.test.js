import React from "react";
import '@testing-library/jest-dom'
import {shallow} from 'enzyme';
import {MultipleCustomHooks} from '../../../component/examples-03/MultipleCustomHooks'
import {useFetch} from '../../../hooks/useFetch'
import {useCounter} from '../../../hooks/useCounter'

jest.mock('../../../hooks/useCounter');
jest.mock('../../../hooks/useFetch');


describe('Pruebas en <MultipleCustomHook>', () => {

    beforeEach (() => {
        useCounter.mockReturnValue({
            counter:10,
            increment: () => {}
    
        })
    });

    
    test('Debe de mostrarse correctamente',() => {

        useFetch.mockReturnValue({
            data:null,
            loading:true,
            error:null

        });
        const wrapper = shallow(<MultipleCustomHooks/>);

        expect(wrapper).toMatchSnapshot();

    });

    test('should debe de mostrar la informacion', () => {

        useFetch.mockReturnValue({
            data: [{
                author: 'Yonatan Pablo',
                quote: 'Desarrollo Web'
            }],
            loading:false,
            error:null
        });

        const wrapper = shallow(<MultipleCustomHooks/>);
        expect(wrapper.find('.alert').exists()).toBe( false );
        expect(wrapper.find('.mb-0').text().trim()).toBe( 'Desarrollo Web' ); 
        expect(wrapper.find('footer').text().trim()).toBe( 'Yonatan Pablo' );        
        console.log(wrapper.html());


    })
})