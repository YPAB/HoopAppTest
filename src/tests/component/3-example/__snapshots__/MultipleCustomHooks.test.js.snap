// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHook> Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Breaking Bad !!
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Cargando ...
  </div>
  <button
    className="btn btn-info"
    onClick={[Function]}
  >
    Siguiente Frase
  </button>
</div>
`;
